Tests/iaas/flavor-naming/flavor-manager-input.py

```diff
@@ -0,0 +1,122 @@
+    sys.exit(rcode)
+
+
+class SpecSyntax:
+    "Abstraction for the output flavor-manager spec file"
+    vocabulary = {"name": "name", "cpus": "0.cpus", "ram": "0.ram*1024", "disk": "1.disksize"}
+
+    def spec_dict():
+        "return dict with syntax specification"
+        ref = [{"field": "name", "mandatory_prefix": "SCS-"},
+               {"field": "public", "default": "true"},
+               {"field": "disabled", "default": "false"}]
+        for key in SpecSyntax.vocabulary:
+            ref.append({"field": key})
+        return {"reference": ref}
+
+    def mand_dict(name, flv):
+        "return a dict for a single flavor, input is the name and the list of properties"
+        dct = {}
+        for key in SpecSyntax.vocabulary:
+            valsel = SpecSyntax.vocabulary[key]
+            if valsel == "name":
+                val = name
+            else:
+                fno, attrcalc = valsel.split('.')
+                attrnm = attrcalc.split("*")[0]
+                try:
+                    val = flv[int(fno)].__getattribute__(attrnm)
+                    if "*" in attrcalc:
+                        val = int(val*int(attrcalc.split("*")[1]))
+                except AttributeError:
+                    val = 0
+            dct[key] = val
+        return dct
+
+
+def main(argv):
+    "main entry point"
+    list_mode = False
+    outfile = sys.stdout
+    scs_mand_file = fnmck.mandFlavorFile
+    errors = 0
+
+    try:
+        opts, args = getopt.gnu_getopt(argv, "hlC:1o:",
+                                       ("help", "list", "mand=", "v1prefer", "outfile="))
+    except getopt.GetoptError as exc:
+        print(f"{exc}", file=sys.stderr)
+        usage(1)
+    for opt in opts:
+        if opt[0] == "-h" or opt[0] == "--help":
+            usage(0)
+        elif opt[0] == "-l" or opt[0] == "--list":
+            list_mode = True
+        elif opt[0] == "-C" or opt[0] == "--mand":
+            scs_mand_file = opt[1]
+        elif opt[0] == "-1" or opt[0] == "--v1prefer":
+            # fnmck.prefer_old = True
+            pass
+        elif opt[0] == "-o" or opt[0] == "--outfile":
+            outfile = open(opt[1], "w", encoding="UTF-8")
+        else:
+            usage(2)
+    if len(args) > 0 and not list_mode:
+        usage(3)
+
+    outspec = SpecSyntax.spec_dict()
+    if list_mode:
+        scs_mand_flavors = args
+        flavor_type = "optional"
+    else:
+        scs_mand_flavors = fnmck.readmandflavors(scs_mand_file)
+        flavor_type = "mandatory"
+
+    olist = []
+    for name in scs_mand_flavors:
+        try:
+            ret = fnmck.parsename(name)
+            assert ret
+            olist.append(SpecSyntax.mand_dict(name, ret))
+        except NameError as exc:
+            print(f"{exc}", file=sys.stderr)
+            errors += 1
+    outspec[flavor_type] = olist
+    # print(outspec, file=outfile)
+    print(yaml.dump(outspec, default_flow_style=False), file=outfile)
```

Comment:
```suggestion
    print(yaml.dump(outspec, default_flow_style=False, sort_keys=False), file=outfile)
```
<br/>
I'd propose to explicitly not sort keys so that `name` is always the first key.

Compare
```
mandatory:
- cpus: 1
  disk: 0
  name: SCS-1V:4
  ram: 4096
- cpus: 1
  disk: 10
  name: SCS-1V:4:10
  ram: 4096
- cpus: 2
  disk: 0
  name: SCS-2V:8
  ram: 8192
- cpus: 2
  disk: 20
  name: SCS-2V:8:20
  ram: 8192
- cpus: 4
  disk: 0
  name: SCS-4V:16
  ram: 16384
- cpus: 4
  disk: 50
  name: SCS-4V:16:50
  ram: 16384
- cpus: 8
  disk: 0
  name: SCS-8V:32
  ram: 32768
- cpus: 8
  disk: 100
  name: SCS-8V:32:100
  ram: 32768
- cpus: 1
  disk: 0
  name: SCS-1V:2
  ram: 2048
- cpus: 1
  disk: 5
  name: SCS-1V:2:5
  ram: 2048
- cpus: 2
  disk: 0
  name: SCS-2V:4
  ram: 4096
- cpus: 2
  disk: 10
  name: SCS-2V:4:10
  ram: 4096
- cpus: 4
  disk: 0
  name: SCS-4V:8
  ram: 8192
- cpus: 4
  disk: 20
  name: SCS-4V:8:20
  ram: 8192
- cpus: 8
  disk: 0
  name: SCS-8V:16
  ram: 16384
- cpus: 8
  disk: 50
  name: SCS-8V:16:50
  ram: 16384
- cpus: 16
  disk: 0
  name: SCS-16V:32
  ram: 32768
- cpus: 16
  disk: 100
  name: SCS-16V:32:100
  ram: 32768
- cpus: 1
  disk: 0
  name: SCS-1V:8
  ram: 8192
- cpus: 1
  disk: 20
  name: SCS-1V:8:20
  ram: 8192
- cpus: 2
  disk: 0
  name: SCS-2V:16
  ram: 16384
- cpus: 2
  disk: 50
  name: SCS-2V:16:50
  ram: 16384
- cpus: 4
  disk: 0
  name: SCS-4V:32
  ram: 32768
- cpus: 4
  disk: 100
  name: SCS-4V:32:100
  ram: 32768
- cpus: 1
  disk: 0
  name: SCS-1L:1
  ram: 1024
- cpus: 1
  disk: 5
  name: SCS-1L:1:5
  ram: 1024
reference:
- field: name
  mandatory_prefix: SCS-
- default: 'true'
  field: public
- default: 'false'
  field: disabled
- field: name
- field: cpus
- field: ram
- field: disk
```

to 

```
reference:
- field: name
  mandatory_prefix: SCS-
- field: public
  default: 'true'
- field: disabled
  default: 'false'
- field: name
- field: cpus
- field: ram
- field: disk
mandatory:
- name: SCS-1V:4
  cpus: 1
  ram: 4096
  disk: 0
- name: SCS-1V:4:10
  cpus: 1
  ram: 4096
  disk: 10
- name: SCS-2V:8
  cpus: 2
  ram: 8192
  disk: 0
- name: SCS-2V:8:20
  cpus: 2
  ram: 8192
  disk: 20
- name: SCS-4V:16
  cpus: 4
  ram: 16384
  disk: 0
- name: SCS-4V:16:50
  cpus: 4
  ram: 16384
  disk: 50
- name: SCS-8V:32
  cpus: 8
  ram: 32768
  disk: 0
- name: SCS-8V:32:100
  cpus: 8
  ram: 32768
  disk: 100
- name: SCS-1V:2
  cpus: 1
  ram: 2048
  disk: 0
- name: SCS-1V:2:5
  cpus: 1
  ram: 2048
  disk: 5
- name: SCS-2V:4
  cpus: 2
  ram: 4096
  disk: 0
- name: SCS-2V:4:10
  cpus: 2
  ram: 4096
  disk: 10
- name: SCS-4V:8
  cpus: 4
  ram: 8192
  disk: 0
- name: SCS-4V:8:20
  cpus: 4
  ram: 8192
  disk: 20
- name: SCS-8V:16
  cpus: 8
  ram: 16384
  disk: 0
- name: SCS-8V:16:50
  cpus: 8
  ram: 16384
  disk: 50
- name: SCS-16V:32
  cpus: 16
  ram: 32768
  disk: 0
- name: SCS-16V:32:100
  cpus: 16
  ram: 32768
  disk: 100
- name: SCS-1V:8
  cpus: 1
  ram: 8192
  disk: 0
- name: SCS-1V:8:20
  cpus: 1
  ram: 8192
  disk: 20
- name: SCS-2V:16
  cpus: 2
  ram: 16384
  disk: 0
- name: SCS-2V:16:50
  cpus: 2
  ram: 16384
  disk: 50
- name: SCS-4V:32
  cpus: 4
  ram: 32768
  disk: 0
- name: SCS-4V:32:100
  cpus: 4
  ram: 32768
  disk: 100
- name: SCS-1L:1
  cpus: 1
  ram: 1024
  disk: 0
- name: SCS-1L:1:5
  cpus: 1
  ram: 1024
  disk: 5
```

IMHO the latter is way nicer to read.

Reply:
Ack.